### PR TITLE
Adding new set of exercises + Organising corpus.

### PR DESCRIPTION
**Pull request by hernoufM**
This PR brings significant changes to the content and organization of the corpus. The following sets of exercises have been added:

- 59 exercises by Simão Melo de Sousa.
- 21 exercises by Hugo Férée.

These exercises are aimed at relatively beginners in OCaml programming who wish to practice and solidify their understanding. These exercises can also serve as a good challenge for experienced users in OCaml programming. While they are primarily designed for beginners to practice and reinforce their understanding, they may still offer interesting problems and opportunities for more experienced users to further enhance their skills and explore advanced concepts. 

Regarding the changes in organization, we would like to highlight the following:

- All new and existing exercises are organized into a large directed graph using the meta field `backward_exercises`.
- The `focus` field has been populated, which now contains several topics that a specific exercise helps to sharpen.

This PR is closely related to a [PR](https://github.com/ocaml-sf/learn-ocaml/pull/554) created for the Learn-ocaml platform in order to better visualize the new organization.

In conclusion, we would like to express our heartfelt gratitude to the contributors, Hugo Férée (@ @hferee) and Simão Melo de Sousa (@smdsousa), for generously providing us with their valuable exercises. These contributions will greatly enrich the corpus and provide a wealth of learning opportunities for beginners and experienced users alike in the field of OCaml programming.

### CODE REVIEW

**Comment by hferee** [comment]
First, thank you for this integration work!

In particular, you had to translate all the exercise statements in english ; but there are still many french artifacts remaining, which makes for a weird mix of languages.
This includes:

- variable names (e.g. "bissextile")
- placeholders ("remplacez-moi") 
- custom error messages (in some `test.ml`)

This is certainly quite a lot more work to fix though.

**Comment by hernoufM** [comment]
Hello, @hferee.
Thanks for your feedback, and again, for such a major contribution!
You are absolutely right, that it makes weird to have some french artifacts remaining within exercises. 
We tried to highlight and translate those parts of the code without translation of which an English-speaking user simply could not complete the exercises. A thorough translation of all variables and function names definitely makes sense, but at the same time it adds the risk that some part of the exercise has not been changed and will lead to an error or incorrect interpretation of the result. Despite all this, a full translation will most likely be reviewed and made in the near future.

**Comment by erikmd** [comment]
Done: https://ocaml-sf.org/learn-ocaml-public/#activity=exercises
